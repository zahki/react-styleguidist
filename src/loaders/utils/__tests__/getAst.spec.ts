import acornJsx from 'acorn-jsx';
import getAst from '../getAst';

describe('getAst', () => {
	test('return AST', () => {
		const result = getAst(`42`);
		expect(result).toMatchSnapshot();
	});

	test('accept Acorn plugins', () => {
		const result = getAst(`<X />`, [acornJsx()]);
		expect(result).toMatchSnapshot();
	});
});
