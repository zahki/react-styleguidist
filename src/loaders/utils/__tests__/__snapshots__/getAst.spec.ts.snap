// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getAst accept Acorn plugins 1`] = `
Node {
  "body": Array [
    Node {
      "end": 5,
      "expression": Node {
        "children": Array [],
        "closingElement": null,
        "end": 5,
        "openingElement": Node {
          "attributes": Array [],
          "end": 5,
          "name": Node {
            "end": 2,
            "name": "X",
            "start": 1,
            "type": "JSXIdentifier",
          },
          "selfClosing": true,
          "start": 0,
          "type": "JSXOpeningElement",
        },
        "start": 0,
        "type": "JSXElement",
      },
      "start": 0,
      "type": "ExpressionStatement",
    },
  ],
  "end": 5,
  "sourceType": "module",
  "start": 0,
  "type": "Program",
}
`;

exports[`getAst return AST 1`] = `
Node {
  "body": Array [
    Node {
      "end": 2,
      "expression": Node {
        "end": 2,
        "raw": "42",
        "start": 0,
        "type": "Literal",
        "value": 42,
      },
      "start": 0,
      "type": "ExpressionStatement",
    },
  ],
  "end": 2,
  "sourceType": "module",
  "start": 0,
  "type": "Program",
}
`;
